src/vanflow.c

```diff
@@ -1532,23 +1539,28 @@ static void _vflow_final(void *adaptor_context)
     //
     // Join and free the thread
     //
-    sys_thread_join(thread);
-    sys_thread_free(thread);
+    sys_thread_join(state->thread);
+    sys_thread_free(state->thread);
 
     //
     // Free the allocated my-address
     //
-    free(event_address_my);
-    free(command_address);
-    free(local_router_id);
+    free(state->event_address_my);
+    free(state->command_address);
+    free(state->local_router_id);
 
     //
     // Free the condition and lock variables
     //
-    sys_cond_free(condition);
-    sys_mutex_free(lock);
-    sys_mutex_free(id_lock);
+    sys_cond_free(state->condition);
+    sys_mutex_free(state->lock);
+    sys_mutex_free(state->id_lock);
+
+    //
+    // Free the module state
+    //
+    free(state);
```

Comment:
for extra re-entrancy Karma, set state = 0; after the free.

Reply:
can't think of a situation where that would be all that helpful; regarding karmic effects I hesitate to speculate

anyways, whatever this is, it's not re-entrancy, even with this you cannot run two routers from the same process _at the same time_ ;P